### DISCUSSION TITLE
What's the difference between LightGBMXT, LightGBM and LightGBMLarge?

### DISCUSSION THREAD

**Discussion by bioinformatica** [autogluon/autogluon#2364] [Q&A] [answered]
1. What's the difference between LightGBMXT, LightGBM and LightGBMLarge?
2. What's the difference between NeuralNetTorch and NeuralNetFastAI. Are they all MLP (Multilayer Perceptron) ?

**Comment by gradientsky** [comment, marked as answer]
1. Different hyperparameters: `LightGBMXT` - enables [extra trees](https://lightgbm.readthedocs.io/en/latest/Parameters.html#extra_trees); `LightGBM` runs with default hyperparameters; `LightGBMLarge` runs with a custom config which enables larger models but trains slower. See the details [here](https://github.com/awslabs/autogluon/blob/3f56287b66f8935f4f19450fd08788878bd40b73/tabular/src/autogluon/tabular/models/lgb/hyperparameters/parameters.py#L30).
2. Yes, they are both MLPs, but different implementations. I would suggest looking into the models code to learn more about the specific details: [fastainn](https://github.com/awslabs/autogluon/tree/3f56287b66f8935f4f19450fd08788878bd40b73/tabular/src/autogluon/tabular/models/fastainn) and [tabular_nn](https://github.com/awslabs/autogluon/tree/3f56287b66f8935f4f19450fd08788878bd40b73/tabular/src/autogluon/tabular/models/tabular_nn).
